test/app/components/market/submit/parameter-input.component.spec.ts

```diff
@@ -40,21 +51,234 @@ fdescribe("ParameterInputComponent", () => {
+            fixture.detectChanges();
+            expect(inputEl.nativeElement.value).toBe(String(updatedInput));
+        });
+
+    });
+
+    describe("dropdown parameter type", () => {
+        let selectEl: DebugElement;
+        let selectComponent: MdSelect;
+        const initialInput: string = "a";
+        const newInput: string = "b";
+
+        beforeEach(() => {
+            testComponent.param = new NcjParameterWrapper("jobName", {
+                type: NcjParameterRawType.string,
+                metadata : {
+                    description: "description",
+                },
+                allowedValues : ["a", "b", "c"],
+                defaultValue: initialInput,
+            });
+            testComponent.paramControl.setValue(initialInput);
+            fixture.detectChanges();
+            selectEl = de.query(By.css("md-select"));
+            expect(selectEl).not.toBeFalsy();
+            selectComponent = selectEl.componentInstance;
+        });
+
+        it("should show all options", () => {
+            const options = selectComponent.options.toArray();
+            expect(options.length).toBe(3);
+            expect(options[0].value).toEqual("a");
+            expect(options[1].value).toEqual("b");
+            expect(options[2].value).toEqual("c");
+        });
+
+        it("should select new input", () => {
+            testComponent.paramControl.setValue(newInput);
+            fixture.detectChanges();
+            expect((selectComponent.selected as MdOption).value).toBe(newInput);
+        });
+    });
+
+    describe("filegroup parameter type", () => {
+        const initialInput: string = "blender-outputs";
+        const newInput: string = "newinput";
+        let fileGroupComponent: FileGroupPickerComponent;
+        let fileGroupEl: DebugElement;
+
+        beforeEach(() => {
+            testComponent.param = new NcjParameterWrapper("outputFileGroup", {
+                type: NcjParameterRawType.string,
+                metadata : {
+                    description: "description",
+                    advancedType: NcjParameterExtendedType.fileGroup,
+                },
+            });
+            testComponent.paramControl.setValue(initialInput);
+            fixture.detectChanges();
+            fileGroupEl = de.query(By.css("bl-file-group-picker"));
+            expect(fileGroupEl).not.toBeFalsy();
+            fileGroupComponent = fileGroupEl.componentInstance;
+        });
+
+        it("should show initial input", () => {
+            expect(fileGroupComponent.value.value).toBe(initialInput);
+        });
+
+        it("should show updated input", () => {
+            testComponent.paramControl.setValue(newInput);
+            fixture.detectChanges();
+            fileGroupComponent = fileGroupEl.componentInstance;
+            expect(fileGroupComponent.value.value).toBe(newInput);
+        });
+    });
+
+    describe("fileinput parameter type", () => {
```

Comment:
for this type you alos want to check the containerId given match what's in the formValues

Reply:
done